packages/dashboard-core-plugins/src/panels/NotebookPanel.jsx

```diff
@@ -489,6 +485,45 @@ class NotebookPanel extends Component {
     this.setState({ isFocused: true });
   }
 
+  /**
+   * @param {MouseEvent} event The click event from clicking on the link
+   */
+  handleLinkClick(event) {
+    const { notebooksUrl, session, sessionLanguage } = this.props;
+    const { href } = event.target;
+    if (!href || !href.startsWith(notebooksUrl)) {
+      return;
+    }
+
+    event.stopPropagation();
+    event.preventDefault();
+
+    const notebookPath = `/${href.substring(notebooksUrl.length)}`.replace(
+      /%20/g,
+      ' '
+    );
+    if (notebookPath === '/') {
```

Comment:
There's also `/existing-dir/`, `/non-existing-dir/` etc.. which are not handled correctly.
Can we somehow check if the link references a file?

Reply:
I'll filter out everything with a trailing slash as well from being transformed